test/executor.test.js

```diff
@@ -906,6 +906,43 @@ describe('executor', function() {
     });
   });
 
+  describe.only('when booting with lazy connect', function() {
+    var DATASOURCE_THROWING_CONNECTION_ERROR = {
+      lazyConnector: {
+        connector: 'testLazyConnect',
+        name: 'lazyConnector',
+      },
+    };
+    var connectTriggered = true;
+
+    beforeEach(function() {
+      app.connector('testLazyConnect', {
+        initialize: function(dataSource, callback) {
+          if (dataSource.settings.lazyConnect) {
+            connectTriggered = false;
+          }
+        },
+      });
+    });
+
+    it('should trigger connect', function(done) {
+      boot.execute(app, someInstructions({
+        dataSources: DATASOURCE_THROWING_CONNECTION_ERROR,
+      }));
+      expect(connectTriggered).to.be.true;
+      done();
+    });
+
+    it('should not trigger connect', function(done) {
+      process.env.LB_LAZYCONNECT_DATASOURCES = true;
+      boot.execute(app, someInstructions({
+        dataSources: DATASOURCE_THROWING_CONNECTION_ERROR,
+      }));
+      expect(connectTriggered).to.be.false;
+      done();
+    });
+  });
+
```

Comment:
Please add unit-tests for the following cases too - notice the value is always a string!

 - `LB_LAZYCONNECT_DATASOURCES='false'`
 - `LB_LAZYCONNECT_DATASOURCES='0'`